orchestrator/test_runner/src/erc_721_happy_path.rs

```diff
@@ -0,0 +1,638 @@
+    // if we don't do this the orchestrators may run ahead of us and we'll be stuck here after
+    // getting credit for two loops when we did one
+    let starting_eth_valset_nonce = get_valset_nonce(gravity_address, *MINER_ADDRESS, web30)
+        .await
+        .expect("Failed to get starting eth valset");
+    let start = Instant::now();
+
+    let (delegate_address, amount) = get_validator_to_delegate_to(contact).await;
+    info!(
+        "Delegating {} to {} in order to generate a validator set update",
+        amount, delegate_address
+    );
+    contact
+        .delegate_to_validator(
+            delegate_address,
+            amount,
+            get_fee(),
+            keys[1].validator_key,
+            Some(TOTAL_TIMEOUT),
+        )
+        .await
+        .unwrap();
+
+    check_valset_update_attestation(&mut grpc_client, keys).await;
+
+    let mut current_eth_valset_nonce = get_valset_nonce(gravity_address, *MINER_ADDRESS, web30)
+        .await
+        .expect("Failed to get current eth valset");
+
+    while starting_eth_valset_nonce == current_eth_valset_nonce {
+        info!(
+            "Validator set is not yet updated to {}>, waiting",
+            starting_eth_valset_nonce
+        );
+        current_eth_valset_nonce = get_valset_nonce(gravity_address, *MINER_ADDRESS, web30)
+            .await
+            .expect("Failed to get current eth valset");
+        delay_for(Duration::from_secs(4)).await;
+        if Instant::now() - start > TOTAL_TIMEOUT {
+            panic!("Failed to update validator set");
+        }
+    }
+    assert!(starting_eth_valset_nonce != current_eth_valset_nonce);
+    info!("Validator set successfully updated!");
+}
+
+// Checks for a MsgValsetUpdatedClaim attestation where every validator is represented
+async fn check_valset_update_attestation(
+    grpc_client: &mut GravityQueryClient<Channel>,
+    keys: &[ValidatorKeys],
+) {
+    let mut found = true;
+    iterate_attestations(grpc_client, &mut |decoded: MsgValsetUpdatedClaim| {
+        // Check that each bridge validator is one of the addresses in our keys
+        for bridge_val in decoded.members {
+            let found_val = keys.iter().any(|key: &ValidatorKeys| {
+                let eth_pub_key = key.eth_key.to_address().to_string();
+                bridge_val.ethereum_address == eth_pub_key
+            });
+            if !found_val {
+                warn!(
+                    "Could not find BridgeValidator eth pub key {} in keys",
+                    bridge_val.ethereum_address
+                );
+            }
+            found &= found_val;
+        }
+    })
+    .await;
+    assert!(
+        found,
+        "Could not find the valset updated attestation we were looking for!"
+    );
+    info!("Found the expected MsgValsetUpdatedClaim attestation");
+}
+
+// Tests an ERC20 deposit and panics on failure
+#[allow(clippy::too_many_arguments)]
+pub async fn test_erc20_deposit_panic(
```

Comment:
this needs to be modified to call your erc721 deposit function

Reply:
Updated: 

```
pub async fn test_erc721_deposit_panic(
    web30: &Web3,
    contact: &Contact,
    dest: CosmosAddress,
    gravity_address: EthAddress,
    gravity_erc721_address: EthAddress,
    erc721_address: EthAddress,
    token_id: Uint256,
    timeout: Option<Duration>, 
)
```